sklearn/compose/_column_transformer.py

```diff
@@ -417,6 +422,20 @@ def _validate_output(self, result):
                     "The output of the '{0}' transformer should be 2D (scipy "
                     "matrix, array, or pandas DataFrame).".format(name))
 
+    def _index_output(self, Xs):
```

Comment:
Suggestion:

```suggestion
    def _record_output_indices(self, Xs):
```

or 

```suggestion
    def _fill_transformers_output(self, Xs):
```

Although I'd also suggest to change the name `transformers_output` to something else